graph/pipeline.go

```diff
@@ -96,7 +146,57 @@ func (p *Pipeline) initialize() {
 			s.CompletedChan = make(chan bool)
 		}
 
+		// Adjust the run command so that the ACR registry is prefixed for all tags
+		s.Run = util.PrefixTags(s.Run, p.RegistryName)
```

Comment:
>PrefixTags [](start = 15, length = 10)

PrefixTags might be able to return Tags directly

Reply:
can't here, because the step's run command will be used directly during build; we have to modify the run command so that that tag names get passed correctly to the scanner + docker

in reality I just added `Tags` and `BuildArgs` to make it easier/"cleaner" to read the node processing logic

Reply:
I see. I thought PrefixTags could return both the prefixed Run string and prefixed Tags array so you don't need to call ParseTags again. I notice ParseTags also does additional work to dedup.